Clean up README.md (#648)
open-telemetry/opentelemetry-dotnet: README.md

# OpenTelemetry .NET

> [![Gitter chat](https://img.shields.io/gitter/room/opentelemetry/opentelemetry-dotnet)](https://gitter.im/open-telemetry/opentelemetry-dotnet)

Reviewer: mention auto-instr channel as well please

Author: I think we want to follow Java model and have a separate repo/SIG?
For example https://github.com/open-telemetry/opentelemetry-auto-instr-java.

Reviewer: It doesn't exist yet as we need to decide how to proceed. I thought to mention this here for now

Author: Given it is not previously mentioned in the doc and we just get started on the auto-instrumentation, I suggest that we make decision on whether we should have a separate repo/SIG next week, and leave it outside this PR.

| LightStep            | [![MyGet Nightly][OpenTelemetry-exporter-lightstep-myget-image]][OpenTelemetry-exporter-lightstep-myget-url]     | [![NuGet release][OpenTelemetry-exporter-lightstep-nuget-image]][OpenTelemetry-exporter-lightstep-nuget-url]     |
| Honeycomb            |                                                                                                                  | [![NuGet release][OpenTelemetry-exporter-honeycomb-nuget-image]][OpenTelemetry-exporter-honeycomb-nuget-url]     |
| NewRelic             |                                                                                                                  | [![NuGet release][OpenTelemetry-exporter-newrelic-nuget-image]][OpenTelemetry-exporter-newrelic-nuget-url]       |
| Console              | [![MyGet Nightly][OpenTelemetry-exporter-console-myget-image]][OpenTelemetry-exporter-console-myget-url]         |                                                                                                                  |

> ## OpenTelemetry Tracing QuickStart: Collecting Data

Reviewer: Can you move this into a separate document for now so it wouldn't be lost while we are in transition

Author: I've moved most of the stuff to specific folder.
The general document on tracing and metrics need to be rewritten and I'll address it in my next PR.

Reviewer: there is also sections on writing custom sampler and exporter. Would you mind just copy it for now as a separate doc, shouldn't be a problem to remove it with the next PR. Just want to keep some docs around.

Author: Good idea, I've moved it to a separate doc under `/docs` folder.

> The .NET [OpenTelemetry](https://opentelemetry.io/) client.

Reviewer: ```suggestion
The .NET [OpenTelemetry](https://opentelemetry.io/) SDK.

```

I think client may be a confusing term here

Author: I remember we've discussed before and the conclusion is to use client? For example:
https://github.com/open-telemetry/opentelemetry-go
https://github.com/open-telemetry/opentelemetry-python

Reviewer: thanks! makes sense

using (tracer.StartActiveSpan("incoming request", out var span))
{
    span.AddEvent("Start processing video.");
    Thread.Sleep(TimeSpan.FromMilliseconds(10));
    span.AddEvent("Finished processing video.");
}
```
### Using Application Insights exporter in Web App

1. Create [Application Insights][ai-get-started] resource.
2. In Startup class, ConfigureServices method add a call of AddOpenTelemetry 
3. Configurate Application Insights inside AddOpenTelemetry with call UseApplicationInsights
4. Set instrumentation key via telemetry configuration object
    telemetryConfiguration.InstrumentationKey = instrumentationKey;
5. See [sample][ai-web-sample] for example use.

``` csharp
 services.AddOpenTelemetry((sp, builder) =>
            {
                    builder.UseApplicationInsights(telemetryConfiguration =>
                    {
                        var instrumentationKey = this.Configuration.GetValue<string>("ApplicationInsights:InstrumentationKey");
                        telemetryConfiguration.InstrumentationKey = instrumentationKey;
                    })
                    .AddRequestAdapter()
                    .AddDependencyAdapter();
            });
```

> ### Using LightStep exporter

Reviewer: Can you please move this to the Lightstep exporter readme.md file

Author: Sure, moved exporter specific docs to exporter folder.

# OpenTelemetry .NET
[![Gitter chat](https://img.shields.io/gitter/room/opentelemetry/opentelemetry-dotnet)](https://gitter.im/open-telemetry/opentelemetry-dotnet)
[![Build Status](https://dev.azure.com/opentelemetry/pipelines/_apis/build/status/open-telemetry.opentelemetry-dotnet-myget-update?branchName=master)](https://dev.azure.com/opentelemetry/pipelines/_build/latest?definitionId=2&branchName=master)

The .NET [OpenTelemetry](https://opentelemetry.io/) client.

## Installation

This repository includes multiple installable packages. The `OpenTelemetry.Api`

> package includes abstract classes and no-op implementations that comprise the OpenTelemetry API following
> [the
> specification](https://github.com/open-telemetry/opentelemetry-specification).

Reviewer 2: ```suggestion
package includes abstract classes and no-op implementations for the [OpenTelemetry API 
specification](https://github.com/open-telemetry/opentelemetry-specification).

```

The `OpenTelemetry` package is the reference implementation of the API.

Libraries that produce telemetry data should only depend on `OpenTelemetry.Api`,
and defer the choice of the SDK to the application developer. Applications may
depend on `OpenTelemetry` or another package that implements the API.

**Please note** that this library is currently in _alpha_, and shouldn't
generally be used in production environments.

The API and SDK packages are available on the following NuGet feeds:
* [MyGet V2](https://www.myget.org/F/opentelemetry/api/v2)
* [MyGet V3](https://www.myget.org/F/opentelemetry/api/v3/index.json)

> To install the development versions of these packages instead, clone or fork this repo and do an editable install:

Reviewer 2: Wouldn't this better on the contributing doc since it is typically used by developers?

Author: Good point.

Author: I've removed this line.
Will send a separate PR to update / clean up the `CONTRIBUTING.md`.

```
TBD
```

## Documentation

The online documentation is available at TBD.

## Compatible Exporters

> See the [OpenTelemetry registry](https://opentelemetry.io/registry/?s=net) for a list of exporters available.

Reviewer 2: We will need a follow up since https://opentelemetry.io/registry/?s=net is not up to date.

Author: Definitely! This can be covered when we clean up the exporters, tracked by #645 #646 #647.
